homeassistant/components/sensor/__init__.py

```diff
@@ -100,10 +101,24 @@ def state_class(self) -> str | None:
         """Return the state class of this entity, from STATE_CLASSES, if any."""
         return None
 
+    @property
+    def last_reset(self) -> datetime | None:
+        """Return the time when the sensor was last reset, if any."""
+        return None
+
     @property
     def capability_attributes(self) -> Mapping[str, Any] | None:
         """Return the capability attributes."""
-        if self.state_class:
-            return {ATTR_STATE_CLASS: self.state_class}
+        if state_class := self.state_class:
```

Comment:
why the need for `state_class` instead of `self.state_class`?

Reply:
it doesn't matter. The only difference that this is now just a single property call (no matter how the property is implemented).